### ISSUE TITLE
[LIT] extensions/private_alloca.cpp failed on post commit

### ISSUE THREAD

**Issue by uditagarwal97** [intel/llvm#13625]
### Describe the bug

Possibly related to https://github.com/intel/llvm/pull/13490

Failed run: https://github.com/intel/llvm/actions/runs/8902870243/job/24449594832

```
FAIL: SYCL :: extensions/private_alloca.cpp (274 of 1021)
******************** TEST 'SYCL :: extensions/private_alloca.cpp' FAILED ********************
Exit Code: 1

Command Output (stdout):
--
# RUN: at line 1
/__w/llvm/llvm/build/bin/clang --driver-mode=g++  -Xsycl-target-backend=amdgcn-amd-amdhsa --offload-arch=gfx906 -fsycl -fsycl-device-only -c -o /__w/llvm/llvm/build/tools/sycl/test/extensions/Output/private_alloca.cpp.tmp.bc /__w/llvm/llvm/src/sycl/test/extensions/private_alloca.cpp
# executed command: /__w/llvm/llvm/build/bin/clang --driver-mode=g++ -Xsycl-target-backend=amdgcn-amd-amdhsa --offload-arch=gfx906 -fsycl -fsycl-device-only -c -o /__w/llvm/llvm/build/tools/sycl/test/extensions/Output/private_alloca.cpp.tmp.bc /__w/llvm/llvm/src/sycl/test/extensions/private_alloca.cpp
# .---command stderr------------
# | clang: warning: argument unused during compilation: '-Xsycl-target-backend=amdgcn-amd-amdhsa --offload-arch=gfx906' [-Wunused-command-line-argument]
# | /__w/llvm/llvm/src/sycl/test/extensions/private_alloca.cpp:52:52: warning: 'legacy' is deprecated: sycl::access::decorated::legacy is deprecated since SYCL 2020 [-Wdeprecated-declarations]
# |    52 |            int, int16_id, sycl::access::decorated::legacy>(kh),
# |       |                                                    ^
# | /__w/llvm/llvm/build/bin/../include/sycl/access/access.hpp:66:10: note: 'legacy' has been explicitly marked deprecated here
# |    66 |   legacy __SYCL2020_DEPRECATED("sycl::access::decorated::legacy "
# |       |          ^
# | /__w/llvm/llvm/build/bin/../include/sycl/detail/defines_elementary.hpp:53:40: note: expanded from macro '__SYCL2020_DEPRECATED'
# |    53 | #define __SYCL2020_DEPRECATED(message) __SYCL_DEPRECATED(message)
# |       |                                        ^
# | /__w/llvm/llvm/build/bin/../include/sycl/detail/defines_elementary.hpp:44:38: note: expanded from macro '__SYCL_DEPRECATED'
# |    44 | #define __SYCL_DEPRECATED(message) [[deprecated(message)]]
# |       |                                      ^
# | 1 warning generated.
# `-----------------------------

# executed command: /__w/llvm/llvm/build/bin/llvm-spirv -o - --to-text /__w/llvm/llvm/build/tools/sycl/test/extensions/Output/private_alloca.cpp.tmp_0.spv
# note: command had no output on stdout or stderr
# executed command: FileCheck /__w/llvm/llvm/src/sycl/test/extensions/private_alloca.cpp -check-prefixes=CHECK-SPV
# .---command stderr------------
# | /__w/llvm/llvm/src/sycl/test/extensions/private_alloca.cpp:97:19: error: CHECK-SPV-DAG: expected string not found in input
# | // CHECK-SPV-DAG: Decorate [[#B0]] Alignment 4
# |                   ^
# | <stdin>:1:1: note: scanning from here
# | 119734787 65792 393230 103 0 
```

### To reproduce

DPC++ commit: 004efa39765c5cad1bde849c753e819b5f69edb9

**Comment by uditagarwal97** [comment]
@victor-eds FYI

**Comment by victor-eds** [comment]
Thanks for pointing this out. I'll address tomorrow. Release vs debug build definitely the issue here

**Comment by victor-eds** [comment]
Fix: #13635
